src/backend/metal/MetalBackend.mm

```diff
@@ -102,19 +118,24 @@ void Present(nxtDevice device) {
         return new RenderPass(this, builder);
     }
     SamplerBase* Device::CreateSampler(SamplerBuilder* builder) {
-        return new Sampler(this, builder);
+        return new Sampler(builder);
     }
     ShaderModuleBase* Device::CreateShaderModule(ShaderModuleBuilder* builder) {
-        return new ShaderModule(this, builder);
+        return new ShaderModule(builder);
     }
     TextureBase* Device::CreateTexture(TextureBuilder* builder) {
-        return new Texture(this, builder);
+        return new Texture(builder);
     }
     TextureViewBase* Device::CreateTextureView(TextureViewBuilder* builder) {
-        return new TextureView(this, builder);
+        return new TextureView(builder);
     }
 
     void Device::TickImpl() {
+        resourceUploader->Tick(finishedCommandSerial);
```

Comment:
About naming these methods Tick, the idea is that the code outside of the class doesn't need to know whether it will free or recycle or even leak allocators. We could make it an interface method, but imho we don't have enough of these to warrant putting them in a vector and adding an interface.